packages/components/src/components/auto-height-wrapper/auto-height-wrapper.jsx

```diff
@@ -1,42 +1,34 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 
-class AutoHeightWrapper extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            height: props.default_height,
-            ref: null,
-        };
-    }
+const AutoHeightWrapper = props => {
+    const [height, setHeight] = React.useState(props.default_height);
+    const [ref, setRef] = React.useState(null);
 
-    setHeight = height => this.setState({ height });
-    setRef = ref =>
-        this.setState({ ref }, () => {
-            this.updateHeight();
-        });
-    updateHeight = () =>
-        this.setHeight(
-            this.state.ref.clientHeight > this.props.default_height
-                ? this.state.ref.clientHeight - (this.props.height_offset || 0)
-                : this.props.default_height
-        );
-    componentDidMount() {
-        window.addEventListener('resize', this.updateHeight);
-    }
+    React.useEffect(() => {
+        window.addEventListener('resize', updateHeight);
+        return () => window.removeEventListener('resize', updateHeight);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    React.useEffect(() => {
+        if (ref) updateHeight();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ref]);
```

Comment:
this behaviour is not same as previous one. Previously, we were calling `updateHeight` in setRef callback. Now this is called immediately after the first render.
If `setRef` is called lazily then with the current behaviour it won't call `updateHeight` function